src/components/LabelMapper/LabelMapper.tsx

```diff
@@ -0,0 +1,30 @@
+import { Label as LabelPF, LabelProps as LabelPropsPF } from '@patternfly/react-core';
+
+import { EmptyStateSymbol } from 'components/EmptyStateSymbol/EmptyStateSymbol';
+
+import { uiLogger } from 'services/uiLogger';
+
+interface ILabelMapperItem {
+  text: string;
+  color: LabelPropsPF['color'];
+}
+
+export type ILabelMapper<Types> = {
+  [key in Types as string]: ILabelMapperItem;
+};
+
+interface ILabelMapperProps {
+  mapper: ILabelMapperItem | undefined;
+}
+
+/**
+ * Represents a Label generated by labelMapper object.
+ *
+ * @param mapper - the mapper object that will be used to generate the label
+ */
+export const LabelMapper = ({ mapper }: ILabelMapperProps) => {
```

Comment:
`LabelMapperItem`
unlike `ProductMilestoneCloseStatusLabelMapper`, which indeed maps something (close result) to a label, this component does not map anything, it just extracts `color` and `text`